src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,328 @@
+            <Button variant="primary" onClick={handleSubmit} disabled={requestingData}>
+              {requestingData ? <Spinner size="sm" animation="border" /> : 'Filter Data'}
+            </Button>
+          </Form>
+        </Col>
+      </Row>
+      <Table responsive>
+        <thead>
+          <tr>
+            {objectProperties.map((property) => (
+              <th key={property}>{property}</th>
+            ))}
+            <th>Timestamp</th>
+          </tr>
+        </thead>
+        <tbody>
+          {orderedData.slice(0, MAX_SHOWN_VALUES).map((data) => (
+            <tr key={data.timestamp}>
+              {objectProperties.map((property) => (
+                <td key={property}>{formatAstarteData(data.value[property])}</td>
+              ))}
+              <td>{new Date(data.timestamp).toLocaleString()}</td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+interface ObjectTableListProps {
+  treeNode: AstarteDataTreeNode<AstarteDatastreamObjectData>;
+}
+
+const ObjectTableList = ({ treeNode }: ObjectTableListProps): React.ReactElement => {
+  const dataTreeLeaves = treeNode.getLeaves();
+  if (dataTreeLeaves.length === 0) {
+    return <p>No data sent by the device.</p>;
+  }
+  return (
+    <>
+      {dataTreeLeaves.map((dataTreeLeaf) => (
+        <ObjectDatastreamTable key={dataTreeLeaf.endpoint} dataTreeNode={dataTreeLeaf} />
+      ))}
+    </>
+  );
+};
+
+interface InterfaceDataProps {
+  interfaceData:
+    | AstarteDataTreeNode<AstartePropertyData>
+    | AstarteDataTreeNode<AstarteDatastreamIndividualData>
+    | AstarteDataTreeNode<AstarteDatastreamObjectData>;
+}
+
+const InterfaceData = ({ interfaceData }: InterfaceDataProps): React.ReactElement => {
+  if (interfaceData.dataKind === 'properties') {
+    return <PropertyTree treeNode={interfaceData as AstarteDataTreeNode<AstartePropertyData>} />;
+  }
+  if (interfaceData.dataKind === 'datastream_individual') {
+    return (
+      <IndividualDatastreamTable
+        treeNode={interfaceData as AstarteDataTreeNode<AstarteDatastreamIndividualData>}
+      />
+    );
+  }
+  return (
+    <ObjectTableList treeNode={interfaceData as AstarteDataTreeNode<AstarteDatastreamObjectData>} />
+  );
+};
+
+export default (): React.ReactElement => {
+  const { deviceId = '', interfaceName = '' } = useParams();
+  const astarte = useAstarte();
+  const [requestingData, setRequestingData] = useState(false);
+  const [fetchedData, setFetchedData] =
+    useState<AstarteDataTreeNode<AstarteDatastreamObjectData>>();
+  const [formAlerts, formAlertsController] = useAlerts();
+  const [data, setData] = useState<InterfaceDataProps['interfaceData'] | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+
+  const fetchData = async (since?: string, to?: string) => {
+    setRequestingData(true);
+    setLoading(true);
+    astarte.client
+      .getDeviceDataTree({
```

Comment:
no need to use a data tree here.
This was usefull when quering all paths of an interface, but if we select the path of the query, we will get a simple array of values

Reply:
I should use ```astarte.client.getDeviceData()``` ?

Reply:
Yes!